app/core/models/ocelot/package.scala

```diff
@@ -44,7 +44,7 @@ package object ocelot {
   val listOfintegerRegex: Regex = s"$commaSeparatedIntsPattern".r
   val anyIntegerRegex: Regex = "^[\\-]?\\d+$".r
   val EmbeddedParameterRegex: Regex = """\{(\d)\}""".r
-  val exclusiveOptionRegex: Regex = "\\[exclusive:([^\\]]+)\\]".r
+  val exclusiveOptionRegex: Regex = "\\[exclusive]".r
```

Comment:
not a valid regex ([] are special and only the first is escaped). can probably replace with a literal as its now a constant

Reply:
findFirstMatchIn does expect a Regex, and it does seem to function properly as a regex too (typing anything that's an invalid regex shows up on intellij as such)

Reply:
you've got away with it, if you want to use [ or ] literally rather than their meaning in a regex, they should be escaped, so it works, but its misleading. What I mean by replace with a literal, is using the features of regular expressions, findFirstMatchIn etc. is no longer needed as its been reduced to simple text match and replace i.e could use contains and replaceAll, replaceAllLiterally on String/StringOps. If you stick with regex, there should be a \\ before the second ]